v6/v6_interface.ml

```diff
@@ -0,0 +1,39 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+let service_name = "v6d"
+let queue_name = ref (Xcp_service.common_prefix ^ service_name)
+let default_sockets_dir = "/var/lib/xcp"
+let default_path = ref (Filename.concat default_sockets_dir service_name)
+let uri () = "file:" ^ !default_path
+
+type debug_info = string
+
+exception Invalid_edition of string
```

Comment:
Would it be easier for a client to handle one exception that carries a sum type?